flake.nix

```diff
@@ -47,45 +44,70 @@
     };
   };
 
-  outputs = inputs@{ self, nixpkgs, haskell-nix, iohk-nix, plutarch, pre-commit-hooks, ... }:
+  outputs = inputs@{ self, nixpkgs, haskell-nix, extra-hackage, iohk-nix, plutarch, pre-commit-hooks, ... }:
     let
-      extraSources = [
-        {
-          src = inputs.cardano-prelude;
-          subdirs = [
-            "cardano-prelude"
-          ];
-        }
-        {
-          src = inputs.cardano-crypto;
-          subdirs = [ "." ];
-        }
-        {
-          src = inputs.flat;
-          subdirs = [ "." ];
-        }
-        {
-          src = inputs.cardano-base;
-          subdirs = [
-            "binary"
-            "cardano-crypto-class"
-          ];
-        }
-        {
-          src = inputs.plutus;
-          subdirs = [
-            "plutus-core"
-            "plutus-ledger-api"
-            "plutus-tx"
-            "prettyprinter-configurable"
-            "word-array"
-          ];
-        }
+      # https://github.com/input-output-hk/haskell.nix/issues/1177
+      nonReinstallablePkgs = [
+        "array"
+        "array"
+        "base"
+        "binary"
+        "bytestring"
+        "Cabal"
+        "containers"
+        "deepseq"
+        "directory"
+        "exceptions"
+        "filepath"
+        "ghc"
+        "ghc-bignum"
+        "ghc-boot"
+        "ghc-boot"
+        "ghc-boot-th"
+        "ghc-compact"
+        "ghc-heap"
+        # "ghci"
+        # "haskeline"
+        "ghcjs-prim"
+        "ghcjs-th"
+        "ghc-prim"
+        "ghc-prim"
+        "hpc"
+        "integer-gmp"
+        "integer-simple"
+        "mtl"
+        "parsec"
+        "pretty"
+        "process"
+        "rts"
+        "stm"
+        "template-haskell"
+        "terminfo"
+        "text"
+        "time"
+        "transformers"
+        "unix"
+        "Win32"
+        "xhtml"
       ];
 
+      myhackages = system: compiler-nix-name: extra-hackage.mkHackagesFor system compiler-nix-name (
+        [
+          "${inputs.flat}"
+          "${inputs.cardano-prelude}/cardano-prelude"
+          "${inputs.cardano-crypto}"
+          "${inputs.cardano-base}/binary"
+          "${inputs.cardano-base}/cardano-crypto-class"
+          "${inputs.plutus}/plutus-core"
+          "${inputs.plutus}/plutus-ledger-api"
+          "${inputs.plutus}/plutus-tx"
+          "${inputs.plutus}/prettyprinter-configurable"
+          "${inputs.plutus}/word-array"
+        ]
+      );
+
```

Comment:
Oh the ergonomics gains are real

Reply:
revolutionary technology really, a whole 500 more lines compared to just using cabal!